### ISSUE TITLE
BUG Automatic install of classes fails

### ISSUE THREAD

**Issue by ngazagna** [benchopt/benchOpt#106]
First, I have corrected in the benchmark solver files of `benchmark_logreg_l2` in my branch [FIX_utils_import](https://github.com/ngazagna/benchmark_logreg_l2/tree/FIX_utils_import)
```python
from benchopt.util import safe_import_context
```
by
```python
from benchopt.utils.safe_import import safe_import_context
```
to solve a first bug which should be raised in an issue in all concerned benchmarks.

Then I still get this error because `lightning` cannot be installed automatically by the `Benchmark` class: 
```bash
$ benchopt run ../benchmarks/benchmark_logreg_l2 
Installing Lightning in None:... failed
/home/nidham/phd/benchOpt/benchopt/benchmark.py:116: UserWarning: Some solvers were not successfully installed, and will thus be ignored. Use 'export BENCHO_RAISE_INSTALL_ERROR=true' to stop at any installation failure and print the traceback.
  warnings.warn(
/tmp/tmp02ch3slw: line 2: benchopt.benchmark.Benchmark: No such file or directory
```

**Comment by tomMoral** [comment]
You should use the master version of the benchmark where I fixed all the import error: https://github.com/benchopt/benchmark_logreg_l2/commit/4fc90c20c5a1a2dc3471601eb8b10a242d7e2335

Also, if you want to see the install error, you can call `BENCHO_RAISE_INSTALL_ERROR=1 benchopt run path/bench/logreg_l2`.

Finally, it seems that you are calling the benchmark with an existing conda env that does not have the latest version of benchopt. You should use the `-r` flag that will reinstall everything to get rid of the `line 2: benchopt.benchmark.Benchmark: No such file or directory` error.

**Comment by tomMoral** [comment]
Ok It was a bug on our side actually.
This should have been fixed by #108
